pulsar-broker/src/main/java/org/apache/pulsar/broker/TransactionMetadataStoreService.java

```diff
@@ -180,6 +180,8 @@ public CompletableFuture<Void> handleTcClientConnect(TransactionCoordinatorID tc
                         // so we should jude the store exist again.
                         if (stores.get(tcId) != null) {
                             completableFuture.complete(null);
+                            tcLoadSemaphore.release();
+                            return;
                         }
```

Comment:
Can we move here out of the semaphore, if the store is already there, we don't need to use Semaphore again?

Reply:
If the store is already there, we can complete the future, and this means that this time handleTcClientConnect has completed. We should do nothing again.